src/compiler/checker.ts

```diff
@@ -3951,15 +3956,22 @@ namespace ts {
             return true;
         }
 
+        // A valid base type is any non-generic object type or intersection of non-generic
+        // object types.
+        function isValidBaseType(type: Type): boolean {

```

Comment:
Well, `BaseType` is the most restrictive type we can express for a valid base type, but there are still `BaseType` instances that aren't valid base types (e.g. intersections containing type parameters). So, wouldn't be correct in the negative sense.